### PR TITLE
chore(TS): BaseBrush abstract methods

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Finalizes #8182

### CODE REVIEW

File: src/brushes/base_brush.class.ts [hunk 1]
```diff
@@ -100,6 +105,11 @@ export abstract class BaseBrush {
     ctx.transform(v[0], v[1], v[2], v[3], v[4], v[5]);
   }
 
+  protected needsFullRender() {
```
Comment (ShaMan123) [inline src/brushes/base_brush.class.ts:108]: moved it because it was in between shadow methods

**Review by ShaMan123 — COMMENTED** [review]
Added abstract methods for clarity and ease of development

File: src/brushes/base_brush.class.ts [hunk 2]
```diff
@@ -75,6 +78,14 @@ export abstract class BaseBrush {
     this.canvas = canvas;
   }
 
+  abstract _render(): void;
+  abstract onMouseDown(pointer: Point, ev: TBrushEventData): void;
+  abstract onMouseMove(pointer: Point, ev: TBrushEventData): void;
+  /**
+   * @returns true if brush should continue blocking interaction
+   */
+  abstract onMouseUp(ev: TBrushEventData): boolean | void;
```
Comment (ShaMan123) [inline src/brushes/base_brush.class.ts:87]: I want to make all signature the same
```suggestion
  abstract onMouseUp(pointer: Point, ev: TBrushEventData): boolean | void;
```

Comment (asturur) [reply src/brushes/base_brush.class.ts:88]: Let's plan the new api directly.

**Review by asturur — DISMISSED** [review]

**Review by asturur — APPROVED** [review]

**Pull request merged**